src/common.rs

```diff
@@ -3391,6 +3391,95 @@ impl Cpu {
     pub fn frequency(&self) -> u64 {
         self.inner.frequency()
     }
+
+    /// Returns the CPU's arch (e.g. x86_64, arm64, ...).
+    ///
+    /// ```no_run
+    /// use sysinfo::{System, RefreshKind, CpuRefreshKind};
+    ///
+    /// let s = System::new_with_specifics(
+    ///     RefreshKind::new().with_cpu(CpuRefreshKind::everything()),
+    /// );
+    /// for cpu in s.cpus() {
+    ///     println!("{}", cpu.arch());
+    /// }
+    /// ```
+    pub fn arch(&self) -> CpuArch {
+        self.inner.arch()
+    }
+}
+
+/// Enumeration of CPU architectures
+#[derive(Eq, PartialEq, Clone, Debug, Copy)]
+pub enum CpuArch {
+    /// `arm` platform
+    ARM,
+    /// `aarch64` or `arm64` platform
+    ARM64,
+    /// `mips` platform
+    MIPS,
+    /// `mips64` platform
+    MIPS64,
+    /// `powerpc` platform
+    POWERPC,
+    /// `powerpc64` platform
+    POWERPC64,
+    /// `riscv32` platform
+    RISCV32,
+    /// `riscv64` platform
+    RISCV64,
+    /// `wasm` platform
+    WASM,
+    /// `wasm64` platform
+    WASM64,
+    /// `x86` platform
+    X86,
+    /// `x64_64` AMD/Intel platform
+    X86_64,
+    /// Unknown platform
+    UNKNOWN,
```

Comment:
Seems like more archs exist (I saw that in [this freebsd man page]()) (only listing the one that aren't already here):

```
aarch64
alpha
amd64
armeb
armv6
armv7
ia64
i386
mipsel
mipselhf
mipshf
mipsn32
mips64el
mips64elhf
mips64hf
pc98
powerpcspe
powerpc64le
riscv64sf
sparc64
```

Not sure how relevant they are though...